### PR TITLE
Fix a check fail in GRUBlockCell

### LINKED ISSUE #58261: A Check Fail can be triggerred in GRUBlockCell
### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.9 and 2.12.0-dev20221018

### Custom Code

No

### OS Platform and Distribution

Linux Ubuntu 20.04

### Mobile device

_No response_

### Python version

3.8

### Bazel version

_No response_

### GCC/Compiler version

_No response_

### CUDA/cuDNN version

CUDA 11.5

### GPU model and memory

_No response_

### Current Behaviour?

```shell
A check fail can be triggerred in GRUBlockCell, which can lead to a crash.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import numpy as np
print(tf.__version__)
for _ in range(20):
    try:
            x = tf.random.uniform([1, 0, 1], dtype=tf.float32)
            h_prev = tf.random.uniform([1, 1, 1], dtype=tf.float32)
            w_ru = tf.random.uniform([1, 2, 1, 1, 1, 1], dtype=tf.float32)
            w_c = tf.random.uniform([1, 1, 1], dtype=tf.float32)
            b_ru = tf.random.uniform([2], dtype=tf.float32)
            b_c = tf.random.uniform([1], dtype=tf.float32)
            res = tf.raw_ops.GRUBlockCell(
                x=x,
                h_prev=h_prev,
                w_ru=w_ru,
                w_c=w_c,
                b_ru=b_ru,
                b_c=b_c,
            )
    except:
        pass
```

### Relevant log output

```shell
F tensorflow/core/framework/tensor_shape.cc:45] Check failed: NDIMS == dims() (2 vs. 3)Asking for tensor of 2 dimensions from a tensor of 3 dimensions
Aborted (core dumped)
```

### PR DESCRIPTION
**Pull request by shogohida**
Signed-off-by: Shogo Hida <shogo.hida@gmail.com>

Fixes #58261

### CODE REVIEW

**Comment by shogohida** [comment]
I'm not sure if the changes I made are correct but I made this PR ready for review so that others can review

File: tensorflow/core/kernels/rnn/gru_ops.cc [hunk 1]
```diff
@@ -121,6 +121,11 @@ class GRUCellBlockOp : public OpKernel {
         ctx, ctx->allocate_output("c", TensorShape({batch_size, cell_size}),
                                   &c_tensor));
 
+    Tensor* g_tensor = nullptr;
+    OP_REQUIRES_OK(
+        ctx, ctx->allocate_output("g", TensorShape({batch_size, cell_size}),
```
Comment (swachhandl) [inline tensorflow/core/kernels/rnn/gru_ops.cc:126]: IIUC, you're creating a new tensor and allocating memory here. I don't think that's needed or would fix the issue. I would imagine you'd have to add checks to the existing Tensors here. 

Please take a look at this PR, which is fixing something similar: https://github.com/tensorflow/tensorflow/pull/59477

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
`g_tensor` is not used at all

**Comment by diego-alayon** [comment]
Unsubscribe

**Comment by mihaimaruseac** [comment]
This has been suprseeded by internal code changes and also #59703

**Pull request closed without merging**